### PR TITLE
GeochemistryUnitConverter utility added, along with unit tests

### PR DESCRIPTION
**Pull request by WilkAndy**
Refs #17155

### CODE REVIEW

**Comment by WilkAndy** [comment]
Hey @loganharbour , i'm looking for a review on this.   Don't despair - the vast majority of the 4000 lines is new tests and re-doing the doco and old tests.

**Comment by WilkAndy** [comment]
Just reminding @loganharbour about this PR

File: modules/geochemistry/include/utils/GeochemistryUnitConverter.h [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+#pragma once
+#include "MooseTypes.h"
+#include "PertinentGeochemicalSystem.h"
+
+/**
+ * Utilities to convert to and from mole units
+ */
+namespace GeochemistryUnitConverter
+{
+enum class GeochemistryUnits
```
Comment (loganharbour) [inline modules/geochemistry/include/utils/GeochemistryUnitConverter.h:19]:
```suggestion
enum class GeochemistryUnit
```

We usually keep these singular - do so if you desire consistency :)

File: modules/geochemistry/include/utils/GeochemistryUnitConverter.h [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+ * @param unit the unit to convert to.  If unit is DIMENSIONLESS,
+ * MOLES or MOLAL, this returns quantity.  If unit is KG, G, MG, UG or CM3, then "moles" is the mole
+ * number.  If unit is KG_PER_KG_SOLVENT, G_PER_KG_SOLVENT, MG_PER_KG_SOLVENT, UG_PER_KG_SOLVENT,
+ * then "moles" is assumed to be the molality.
+ * @param species_name the name of the substance - this should appear in mgd
+ * @param mgd the database corresponding to the model, which should contain information about
+ * species_name
+ */
+Real fromMoles(Real moles,
+               const GeochemistryUnits & unit,
```
Comment (loganharbour) [inline modules/geochemistry/include/utils/GeochemistryUnitConverter.h:51]:
```suggestion
               const GeochemistryUnits unit,
```

FYI with something like this enum that is an int, having this as a reference could be slower! There's a few others below.

**Review by loganharbour — COMMENTED** [review]
Looks good to me. Two small things that you could change if you desire.

We really do need a general unit framework somewhere in the framework.... but it doesn't exist yet!

File: modules/geochemistry/include/utils/GeochemistryUnitConverter.h (hunk 1 above)
Comment (WilkAndy) [reply modules/geochemistry/include/utils/GeochemistryUnitConverter.h:19]: Sure, done.

File: modules/geochemistry/include/utils/GeochemistryUnitConverter.h (hunk 2 above)
Comment (WilkAndy) [reply modules/geochemistry/include/utils/GeochemistryUnitConverter.h:51]: Aha, i get it, thanks.

**Comment by WilkAndy** [comment]
Hey @loganharbour - i've implemented your changes.  Let's hope we see all green soon.

**Review by loganharbour — APPROVED** [review]

**Comment by WilkAndy** [comment]
Thanks @loganharbour for the review and merge

**Pull request merged**